RelationalAI/IAccessTokenHandler.cs

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 RelationalAI, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace RelationalAI
+{
+    // IAccessTokenHandler is a contract interface
+    // for custom access token handlers implementation.
+    // Check DefaultAccessToken for an example
+    public interface IAccessTokenHandler
+    {
+        Rest Rest { get; set; }
```

Comment:
The Rest client should not be part of the interface, nor the Logger. The interface is to simply provide token retrieval.  Baking it into the interface conflates 1) The fact of even using the Rest client to do the implementation and 2) That the implementation logs at a ll, much less using the MS Logging extension.